### ISSUE TITLE
popper.js dependency seems incompatible with Locker Service

### ISSUE THREAD

**Issue by MacMorgan** [salesforce/design-system-react#1806]
When trying to interact with any components with a positioned dropdown (combobox, date picker, etc.) we are getting the following error which originates from findCommonOffsetParent in popper.js. Seems to be from popper.js trying to access properties/methods on the SecureDocument & SecureElement objects.

![screen shot 2019-03-06 at 10 25 48 am](https://user-images.githubusercontent.com/17681/53892585-92df6a80-3ffa-11e9-8c28-856a96fd8b7b.png)

Anyone have any experience with this? I would prefer to continue to use the Locker Service vs. Lightning Container.

**Comment by interactivellama** [comment]
Popper does a search up it's DOM ancestors to see if there are any overflow hidden parents and whether the Popover or men is hitting the viewport edge. Could that be causes the issue?

For Portals (second React ender trees) when `position: overFlowBoundarElement` is used (since you need position like this in for instance, a Modal), the `body` node is used. Some of the components have a `portalMount` callback that change the location of the second render tree in the DOM.

In summary, positioning items on the screen without full access to the Window DOM is tricky. I'm open to ideas if you have any. `hasStaticAligment:true` might help--that would cause menus to not flip which is a more poor UX experience, but might prevent Popper from accessing `window` or other globals.

**Comment by MacMorgan** [comment]
Did a significant amount of research on this. Looks like Locker Service does not properly implement Range.setStart & Range.setEnd which popper.js relies on. A similar issue was fixed in a patch to the Spring 17 Release related to Node APIs (https://salesforce.stackexchange.com/questions/170700/summer-17-lockerservice-error-while-using-node-contains) so I created a similar issue for the Range methods (https://salesforce.stackexchange.com/questions/252824/lockerservice-error-while-using-range-setstart).

Seems the easiest path forward short term is to set the menuPosition to 'relative' and short circuit popper.js entirely. This will require some modifications to the UI for components that previously relied on position: overFlowBoundaryElement.

Will follow up if I get a reply on a fix to the Locker Service to address this issue.

**Comment by interactivellama** [comment]
Sounds good. Thanks!

I guess you could hack/polyfill `popper.js` module at build, while you are waiting, too. It's not built into the library.

**Comment by interactivellama** [comment]
@MacMorgan Did this get worked out with LockerService?
